### ISSUE TITLE
Issues importing players for MLB 

### ISSUE THREAD

**Issue by McCampy** [DimaKudosh/pydfs-lineup-optimizer#281]
Hello all, I have been having a few issues loading players as objects into the optimizer for Fanduel MLB. Using the CSV from Fanduel, I can import players with the load_players_from_csvs method just fine and build lineups with no problems, however,  the load_players method seems to end in an error.  The list of players is correct and is populated with the same amount of players passed in the CSV. 

### Error 
![error1](https://user-images.githubusercontent.com/78242592/126081191-2a8ee72c-4577-4f33-9f2a-698afeb79b9f.png)

### Code
![code](https://user-images.githubusercontent.com/78242592/126081197-ce1a9f40-852c-419a-b208-87735d543f59.png)

Additionally, when I have tried to pass in the game day CSV from rotogrinders, I received a similar error mentioned over in thread #274.  
![error2](https://user-images.githubusercontent.com/78242592/126081203-13d093e9-5275-4d35-82eb-5d3ede0f5099.png)

I appreciate any ideas, thank you 😄

**Comment by DimaKudosh** [comment]
Hi, according to not valid constraints it looks like a problem with players positions. Check that all positions are correct and it should be a list of strings with positions names.

**Comment by McCampy** [comment]
@DimaKudosh So it appears I forgot the brackets around positions=[row.Position} which fixed the load in problem for the Fanduel CSV. I am still getting the same SolverInfeasibleSolutionException and GenerateLineupException error when I attempt to load in players from anywhere else besides the Fanduel CSV. I brought in the positions from the Fanduel CSV and merged them on player name with my other CSV to ensure the positions were correct and unfortunately got the same results. The list of players is being populated correctly as well so I'm not quite sure why it isn't working. Thank you for the help.

**Comment by DimaKudosh** [comment]
Could you provide a sample of your file and code?

**Comment by McCampy** [comment]
Here is a sample of the file from rotogrinders with only 50 players, please let me know if you need the full file. Every column is from rotogrinders except the last column named FD_Position which is from the Fanduel file itself add is merged in a previous data cleaning script. [rotogrinders_projections.csv](https://github.com/DimaKudosh/pydfs-lineup-optimizer/files/6852867/rotogrinders_projections.csv)

The code:
```
import pandas as pd
from pydfs_lineup_optimizer import get_optimizer, Site, Sport, Player
    
optimizer = get_optimizer(Site.FANDUEL, Sport.BASEBALL) 

df = pd.read_csv(r'E:\DFS\CSV_Downloads\07-18-21\rotogrinders_projections.csv')

def row_to_player(row):
    """Converts dataframe row to Player object
    
    Args:
        row (namedtuple): dataframe row
    
    Returns:
        Player
    """
    valid_kwargs = {
        'is_injured', 'max_exposure', 'min_exposure', 
        'projected_ownership', 'game_info', 'roster_order', 
        'min_deviation', 'max_deviation', 'is_confirmed_starter', 
        'fppg_floor', 'fppg_ceil'
    }

    kwargs = {}
    for name, value in row._asdict().items():
        if name in valid_kwargs:
            kwargs[name] = value    
    
    return Player (
            player_id=getattr(row, 'id') if hasattr(row, 'id') else row.first_name + ' ' + row.last_name, 
            first_name=row.first_name,
            last_name=row.last_name,
            positions=[row.FD_Position],
            team=row.team,
            salary=row.salary,
            fppg=row.fpts,
            **kwargs
    )

players = [row_to_player(row) for row in df.itertuples(index=False)]
optimizer.load_players(players)

for lineup in optimizer.optimize(n=1):
```

**Comment by DimaKudosh** [comment]
Your code with csv sample works for me. I see only one problem with your code: you incorrectly parse multipositional players you should make split before setting position: `positions=row.FD_Position.split('/'),`

**Comment by McCampy** [comment]
I'll make sure to fix that. That's very strange that it is working for you and not on my computer. I downloaded their file again today to give it another shot but didn't have any success. I decided to aggregate projections from a different source and loaded those into the optimizer and it ran just fine. I am going to go ahead and close this thread because the main problem was solved and this minor one appears to be on my end. Thanks again for the help.
